py/picca/tests/delta_extraction/expected_flux_tests.py

```diff
@@ -381,69 +572,318 @@ def test_dr16_expected_flux_compute_var_stats(self):
         # compute variance functions and statistics
         expected_flux.compute_var_stats(data.forests)
 
+        # save results
+        f = open(out_file, "w")
+        f.write("#log_lambda eta var_lss fudge num_pixels valid_fit\n")
+        for log_lambda in expected_flux.log_lambda_var_func_grid:
+            f.write(f"{log_lambda} ")
+            f.write(f"{expected_flux.get_eta(log_lambda)} ")
+            f.write(f"{expected_flux.get_var_lss(log_lambda)} ")
+            f.write(f"{expected_flux.get_fudge(log_lambda)} ")
+            f.write(f"{expected_flux.get_num_pixels(log_lambda)} ")
+            f.write(f"{expected_flux.get_valid_fit(log_lambda)} ")
+            f.write("\n")
+        f.close()
+
         # load the expected results
         expectations = np.genfromtxt(test_file, names=True)
 
         # compare with obtained results
         eta = expected_flux.get_eta(expectations["log_lambda"])
         var_lss = expected_flux.get_var_lss(expectations["log_lambda"])
         fudge = expected_flux.get_fudge(expectations["log_lambda"])
+        num_pixels = expected_flux.get_num_pixels(expectations["log_lambda"])
+        valid_fit = expected_flux.get_valid_fit(expectations["log_lambda"])
         self.assertTrue(np.allclose(eta, expectations["eta"]))
         self.assertTrue(np.allclose(var_lss, expectations["var_lss"]))
         self.assertTrue(np.allclose(fudge, expectations["fudge"]))
-
+        self.assertTrue(np.allclose(num_pixels, expectations["num_pixels"]))
+        self.assertTrue(np.allclose(valid_fit, expectations["valid_fit"]))
         # setup Forest variables; case: linear wavelength solution
         reset_forest()
         setup_forest("lin")
-        #TODO: compute_var_stats: add linear wavelength solution test
 
-    def test_dr16_expected_flux_populate_los_ids(self):
-        """Test method populate_los_ids for class Dr16ExpectedFlux"""
-        # setup Forest variables; case: logarithmic wavelength solution
-        setup_forest("log", rebin=3)
+        out_file = f"{THIS_DIR}/results/var_stats_lin.txt"
+        test_file = f"{THIS_DIR}/data/var_stats_lin.txt"
 
         # initialize Data and Dr16ExpectedFlux instances
         config = ConfigParser()
         config.read_dict({
-            "data": sdss_data_kwargs,
+            "data": desi_healpix_kwargs,
```

Comment:
wouldn't it make more sense to do this test on SDSS data (or on both)? Given that in the end for DESI ppl will probably use the lin mode more? But testing DESI with log is useful for cross checking results of lin vs log quickly...